### PR TITLE
refactor: replace deprecated template provider usage

### PR DESCRIPTION
**Pull request by rosstimson**
In Terraform v0.12 and above the `terraform_file` resource that is part
of the template provider has been replaced with a built-in function
`templatefile` that is simpler to use.  You can see this mentioned in
the docs at:
https://registry.terraform.io/providers/hashicorp/template/latest/docs/data-sources/file

You can also see that the `template` provider itself has been
archived: https://github.com/hashicorp/terraform-provider-template
There will therefore be no further updates or builds for it which
becomes a problem for Mac users on the new Apple Silicon chips, it
means that wherever the old and deprecated `template` provider is used
anyone on an Apple machine with M1 chip will not be able to
successfully run `terraform init`.

I'm not sure if there is a better way to iterate over all the
variables that need to be passed into the `templatefile` function as a
map.

### CODE REVIEW

**Comment by nimrodkor** [comment]
LGTM, @JamesWoolfenden PTAL

File: bridgecrew_sns_resource.tf [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
+  external_id          = random_uuid.external_id.result
+  role_arn             = aws_iam_role.bridgecrew_account_role.arn
+  region               = data.aws_region.region.id
+  api_token            = var.api_token
 }
 
 resource "null_resource" "create_bridgecrew" {
   provisioner "local-exec" {
-    command     = "aws sns ${local.profile_str} --region ${data.aws_region.region.id} publish --target-arn \"${local.bridgecrew_sns_topic}\" --message '${jsonencode(data.template_file.message.rendered)}' && sleep 30"
+    command     = "aws sns ${local.profile_str} --region ${data.aws_region.region.id} publish --target-arn \"${local.bridgecrew_sns_topic}\" --message '${jsonencode(templatefile("${path.module}/message.json.tpl", { request_type = local.request_type, bridgecrew_sns_topic = local.bridgecrew_sns_topic, customer_name = local.customer_name, account_id = local.account_id, external_id = local.external_id, role_arn = local.role_arn, region = local.region, api_token = local.api_token }))}' && sleep 30"
```
Comment (nimrodkor) [inline bridgecrew_sns_resource.tf]: @rosstimson Can we save the result of `templatefile()` as a local var, and then reuse it across this file?

**Review by nimrodkor — COMMENTED** [review]
Looks really good, tnx @rosstimson !
Just 1 comment 🙏

File: bridgecrew_sns_resource.tf (hunk 1 above)
Comment (rosstimson) [reply bridgecrew_sns_resource.tf]: I hope https://github.com/bridgecrewio/terraform-aws-bridgecrew-read-only/pull/8/commits/16a53005aa4c28d4a1c45ada24fe5abb76e540dd is what you were after there.  It seems to work for me but will need some further testing.

**Review by JamesWoolfenden — APPROVED** [review]
LGTM

**Pull request merged**